src/url.rs

```diff
@@ -100,6 +113,22 @@ impl Url {
             }
         };
 
+        let alternates: Option<Vec<Alternate>> = match alternates {
+            None => None,
+            Some(alternates) => {
+                let mut unique_hreflangs = HashSet::new();
+                for alternate in &alternates {
+                    if !unique_hreflangs.insert(&alternate.hreflang) {
+                        return Err(UrlError::DuplicateAlternateHreflangs(
```

Comment:
Instead of manually checking for uniqueness within a Vec, could `pub alternates: Option<Vec<Alternate>>,` be refactored to be `pub alternates: Option<Set<Alternate>>,` from the beginning?

Reply:
Going to take a look at that.